docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+| `c`ases     | List(String)                        | Event types                                                    |
+| `u`nfolds   | List(Map([Unfold](#unfold-schema))) | [Unfolds](#unfold-schema)                                      |
+| `a`ppends   | Number                              | The number of events appended in the latest update             |
+| `n`umber    | Number                              | The version of the stream                                      |
+
+<h2 id="event-schema">Event schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+| `x`actionId     | ?String | The correlation id of the event                              |
+| wh`y`           | ?String | The causation id (wh`y` did this event occur)                |
+
+<h2 id="unfold-schema">Unfold schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `i`ndex         | Number  | The index at which the unfold was generated                  |
+| `c`ase          | String  | The event type of the unfold                                 |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+
+# Reactions
+
+Reactions are an often necessary part of event-sourced architectures. DynamoDB
+offers DynamoDB Streams, a Change Data Capture (CDC) for your table. This CDC
+feed offers two guarantees
+
+- Each stream record appears exactly once in the stream
+- For each item that is modified in a DynamoDB table, the stream records appear
+  in the same sequence as the actual modifications to the item.
+- The stream contains the last 24 hours of updates
+
+You'll notice that the second guarantee notably does not guarantee ordering
+across items within the same hash key, only for updates to a singular (hash,
+range) pair. For this reason every event written to DynamoStore will be first
+written to the Tip document.
```

Comment:
will -> must.

Was trying to formulate a sentence but failed.... ~Maybe add "; if we skip 'feeding it in through the tip', the record of inserts of calf items can emanate from a DDB Streams feed after records relating to appends to the tip that preceded it in terms of when the writes actually happened"~

Also you have not made any reference to the key tenets of an event store's projections/reactions interface:-
1. all events for a stream absolutely must be delivered in order with no gaps (at least once delivery can always happen, but a feed that can ever supply you event 3 of a stream before it has presented you with event 2 is worse than useless)
2. there must be a 100% guarantee of all events getting delivered. The alternative is programming by coincidence, littering it with nonsensical defensive 'just in case' code, and spending hours checking logs to see if by any chance some event just might have been missed 'for any reason'.